packages/node_modules/@webex/plugin-meetings/test/unit/spec/meetings/index.js

```diff
@@ -0,0 +1,321 @@
+    beforeEach(() => {
+      spark = new MockSpark({
+        children: {
+          device: Device,
+          mercury: Mercury,
+          meetings: Meetings
+        }
+      });
+    });
+    it('has a spark instance with a meetings property', () => {
+      assert.exists(spark, 'spark was initialized with children');
+      assert.exists(spark.meetings, 'meetings child was set up on the spark instance');
+    });
+    describe('Public API Contracts', () => {
+      describe('gets', () => {
+        describe('#getReachability', () => {
+          describe('before #setReachability', () => {
+            it('does not get a reachability instance', () => {
+              const reachability = spark.meetings.getReachability();
+              assert.notExists(reachability, 'reachability is undefined because #setReachability has not been called');
+            });
+          });
+          describe('after #setReachability', () => {
+            beforeEach(() => {
+              spark.meetings.setReachability();
+              const reachabilityMocker = spark.meetings.getReachability();
+              sinon.stub(reachabilityMocker, 'gatherReachability').returns(true);
+            });
+            it('gets the reachability data instance from spark.meetings', () => {
+              const reachability = spark.meetings.getReachability();
+              assert.exists(reachability, 'reachability is defined because #setReachability has been called');
+              assert.typeOf(reachability, 'object', 'should be an object');
+              assert.instanceOf(reachability, Reachability, 'should be a reachability instance');
+            });
+          });
+        });
+        describe('#getPersonalMeetingRoom', () => {
+          it('gets the personal meeting room instance from spark.meetings', () => {
+            const personalMeetingRoom = spark.meetings.getPersonalMeetingRoom();
+            assert.exists(personalMeetingRoom, 'personal meeting room instance is set up at object creation');
+            assert.typeOf(personalMeetingRoom, 'object', 'should be an object');
+            assert.instanceOf(personalMeetingRoom, PersonalMeetingRoom, 'should be a personal meeting room instance');
+          });
+        });
+        describe('Static shortcut proxy methods', () => {
+          let staticMeetings;
+          beforeEach(() => {
+            staticMeetings = Meetings;
+          });
+          afterEach(() => {
+            staticMeetings = null;
+          });
+          describe('MeetingCollection getByKey proxies', () => {
+            beforeEach(() => {
+              MeetingCollection.getByKey = sinon.stub().returns(true);
+            });
+            describe('#getMeetingByType', () => {
+              it('gets the Meeting instance from MeetingCollection using type and value', () => {
+                staticMeetings.getMeetingByType('test', 'test');
+                assert.calledOnce(MeetingCollection.getByKey);
```

Comment:
Instead of asserting that in internal function was called, we should assert that `test` was returned. I would prefer to insert an actual object (multiple actually) in the `beforeEach` then assert that the correct object is returned.

Reply:
okay, will take a look.